### ISSUE TITLE
Ability to see Glucose Change in Carbohydrates screen prior to current day.

### ISSUE THREAD

**Issue by natetr** [LoopKit/Loop#681]
Is there a way to adjust how far back the carbohydrates screen shows the carbs entered on the chart. When meals are still being absorbed overnight it would be nice to see how the carbs acted when I wake up in the morning. Currently, it looks like Loop resets the carb chart sometime in the middle of the night so I don't see how a meal I ate the prior evening was observed vs. predicted. This would be helpful to know in order to make adjustments to my meals. 
![img_3047](https://user-images.githubusercontent.com/34628512/36355895-2af52fd6-149f-11e8-97da-c571955b2e4d.PNG)

**Comment by elnjensen** [comment]
I agree that this would be useful.  Nightscout helps some here, but doesn't provide the same info that the Loop carb view does. I'm not sure what a good UI solution would be, though.  Is there a way to programmatically snapshot and (temporarily) save what a previous UI view would have been even if the user didn't load that view?

**Comment by natetr** [comment]
Is it possible to just set the time frame to show more hours on the table? I played around with the file "CarbAbsorptionViewController.swift" in this section by increasing max hours to line 109:

`let date = Date(timeIntervalSinceNow: -TimeInterval(hours: max(1, totalHours)))`

While it did allow me to look further back in time it also adjusted the chart at the top to the point that it was unreadable. I don't know Swift (or any code for that matter) so I wasn't sure what else I could adjust to show more hours in just the table.

**Comment by Kdisimone** [comment]
Actually, looking at it a little deeper, I'm guessing you need to go to LoopKit (not Loop).  That looks like the Loop is for the carb absorption (top graph of ICE), not the carbstore.  I'd go to LoopKit.

<img width="1512" alt="screen shot 2018-02-26 at 10 05 08 am" src="https://user-images.githubusercontent.com/12402160/36686857-a97a7aa8-1adc-11e8-9e6f-c13660264c4f.png">

**Comment by natetr** [comment]
@Kdisimone Thanks for finding that for me. I'll see what I can find in there and hopefully I won't break everything :)

**Comment by francescaneri** [comment]
@natetr did you find a solution?

**Comment by natetr** [comment]
I did not, unfortunately. I still think it could be really helpful but I just didn't have the time/know how to dig deeper into it.

**Comment by bharat** [comment]
If you set midnight to an earlier time, it seems to work properly. For example, I set it to a rolling 24 window with this change in one of my branches: https://github.com/bharat/Loop/commit/81b9ef71090d32645a90760b1490ee1b38201803

Here's what it looks like:
![screen shot 2018-06-14 at 1 56 04 pm](https://user-images.githubusercontent.com/54570/41437996-dc7d835c-6fda-11e8-9ceb-0bcbe940d66f.png)

Note that the chart time window doesn't change- that would make the chart too cramped. But it does show you more historical data. If this solves your problem, I'll submit a pull request. Let me know!

**Comment by natetr** [comment]
That's perfect! Thanks for figuring that out. Solves my problem.

**Comment by NightScoutForSeb** [comment]
Showing entries from the last 12h or even a rolling 24h would solve the problem for me.
